src/jit/dll/CMakeLists.txt

```diff
@@ -1,5 +1,9 @@
 project(ClrJit)
 
+if(CLR_CMAKE_PLATFORM_ARCH_I386)
```

Comment:
It may be a good idea to add this to jit/crossgen/CMakeLists.txt too so that both runtime and crossgen have the same flavor of the JIT statically linked in.